src/server/auth.ts

```diff
@@ -16,40 +17,61 @@ import { db } from "@/server/db";
  * @see https://next-auth.js.org/getting-started/typescript#module-augmentation
  */
 declare module "next-auth" {
-  interface Session extends DefaultSession {
+  interface Session {
     user: {
       id: string;
-      // ...other properties
-      // role: UserRole;
+      isOnboarded: boolean;
     } & DefaultSession["user"];
   }
-
-  // interface User {
-  //   // ...other properties
-  //   // role: UserRole;
-  // }
 }
 
+declare module "next-auth/jwt" {
+  interface JWT {
+    id: string;
+    isOnboarded: boolean;
+  }
+}
 /**
  * Options for NextAuth.js used to configure adapters, providers, callbacks, etc.
  *
  * @see https://next-auth.js.org/configuration/options
  */
 export const authOptions: NextAuthOptions = {
   callbacks: {
-    session: ({ session, user }) => ({
-      ...session,
-      user: {
-        ...session.user,
-        id: user.id,
-      },
-    }),
+    async jwt({ trigger, user: adapterUser, token }) {
+      if (trigger) {
+        const user = await db.user.findFirstOrThrow({
+          where: { id: adapterUser.id },
+          select: {
+            completedOnboarding: true,
```

Comment:
Lets keep them consistent, `completedOnboarding` vs `isOnboarded`. I would go with `isOnboarded` everywhere, both on session and column.

Reply:
Since this was part of my older commits, I abandoned it due to some concerns regarding the database schema. Since we are moving towards a multi-tenant route, `isOnboarded` should be part of the `Member` model. When the user is first created, there isn't any organization to create a membership for. should we create a dummy organization and keep it behind a flag and replace with onboarded data ?